docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md

```diff
@@ -180,457 +180,1595 @@ RETURN example.func_example("first argument");
-Bases: `_mgp.`
+- `edge` - `Edge` that is added to the path
+  
 
-Signals that a key already exists in a container-like object.
+**Raises**:
 
+- `InvalidContextError` - If using an invalid `Path` instance or if passed in `Edge` is invalid.
+- `LogicErrorError` - If the current last vertex in the path is not part of the given edge.
+- `UnableToAllocateError` - If unable to allocate memory for path extension.
+  
 
-## class mgp.LogicErrorError(*args: Any, **kwargs: Any)
-Bases: `_mgp.`
+**Examples**:
 
-Signals faulty logic within the program such as violating logical preconditions
-or class invariants and may be preventable.
+  ```path.expand(edge)```
 
+### vertices
 
-## class mgp.OutOfRangeError(*args: Any, **kwargs: Any)
-Bases: `_mgp.`
+```python
+@property
+def vertices() -> typing.Tuple[Vertex, ...]
+```
 
-Signals that an index-like parameter has a value that is outside its possible
-values.
+Vertices are ordered from the start to the end of the path.
 
+**Returns**:
 
-## class mgp.SerializationError(*args: Any, **kwargs: Any)
-Bases: `_mgp.`
+  A tuple of `Vertex` objects order from start to end of the path.
+  
 
-Signals serialization error caused by concurrent modifications from different
-transactions.
+**Raises**:
 
+- `InvalidContextError` - If using an invalid Path instance.
+  
 
-## class mgp.UnableToAllocateError(*args: Any, **kwargs: Any)
-Bases: `_mgp.`
+**Examples**:
 
-Signals failed memory allocation.
+  ```path.vertices```
 
+### edges
 
-## class mgp.UnknownError(*args: Any, **kwargs: Any)
-Bases: `_mgp.`
+```python
+@property
+def edges() -> typing.Tuple[Edge, ...]
+```
 
-Signals unspecified failure.
+Edges are ordered from the start to the end of the path.
 
+**Returns**:
 
-## class mgp.ValueConversionError(*args: Any, **kwargs: Any)
-Bases: `_mgp.`
+  A tuple of `Edge` objects order from start to end of the path
 
-Signals that the conversion failed between python and cypher values.
+**Raises**:
+
+- `InvalidContextError` - If using an invalid `Path` instance.
+
+**Examples**:
+
+  ```path.edges```
+
+## Record Objects
+
+```python
+class Record()
+```
+
+Represents a record of resulting field values.
+
+### \_\_init\_\_
+
+```python
+def __init__(**kwargs)
+```
+
+Initialize with name=value fields in kwargs.
+
+## Vertices Objects
+
+```python
+class Vertices()
+```
+
+Iterable over vertices in a graph.
+
+### is\_valid()
+
+```python
+def is_valid() -> bool
+```
+
+Check if `Vertices` is in valid context and may be used.
+
+**Returns**:
+
+  A `bool` value depends on if the `Vertices` is in valid context.
+  
+
+**Examples**:
+
+  ```vertices.is_valid()```
+
+### \_\_iter\_\_
+
+```python
+def __iter__() -> typing.Iterable[Vertex]
+```
+
+Iterate over vertices.
+
+**Returns**:
+
+  Iterable list of `Vertex` objects.
+  
+
+**Raises**:
+
+- `InvalidContextError` - If context is invalid.
+- `UnableToAllocateError` - If unable to allocate an iterator or a vertex.
+  
+
+**Examples**:
```

Comment:
These examples are written differently than all the others.. not very important but FYI
They should be:
 ```  for vertex in graph.vertices:  ```
 ```  iter(graph.vertices)  ```